packages/core/core-v2/src/Parcel.js

```diff
@@ -112,72 +108,74 @@ export default class Parcel {
     }
   }
 
-  async updateGraph() {
-    // Process all changes that may update the middle of the the graph
-    this.updateQueue.start();
-    await this.updateQueue.onIdle();
-    this.updateQueue.pause();
+  async updateGraph({signal}: BuildOpts) {
+    for (let [id, node] of this.graph.invalidNodes) {
+      this.processNode(node, {signal, shallow: true});
+    }
+
+    await this.queue.run();
   }
 
-  async completeGraph({ signal }: BuildOpts) {
-    // AssetGraph keeps track of nodes that represent files that have not been transformed
-    // and dependencies that have not been resolved so that rebuilds start from the leaves
-    // of the graph
+  async completeGraph({signal}: BuildOpts) {
     for (let [id, node] of this.graph.incompleteNodes) {
-      this.mainQueue.add(() => this.processNode(node, { signal }));
+      this.processNode(node, {signal});
```

Comment:
I actually kinda liked having the actual queuing here instead of in the `resolve`/`transform` methods. It was easier to understand. At first I was going to comment that you missed an `await` here before reading the other methods and realizing that they changed to queue instead of actually resolve or transform.